app/init/network_manager.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Alert} from 'react-native';
+import DeviceInfo from 'react-native-device-info';
+
+import Emm from '@mattermost/react-native-emm';
+import {getOrCreateAPIClient} from '@mattermost/react-native-network-client';
+
+import ManagedApp from '@app/init/managed_app';
+import LocalConfig from '@assets/config';
+import {Client} from '@client/rest';
+import * as ClientConstants from '@client/rest/constants';
+
+import type {ServerCredential} from '@typings/credentials';
+
+// TODO: Should these constants be part of react-native-network-library?
+const CLIENT_CERTIFICATE_IMPORT_ERROR_CODES = [-103, -104, -105, -108];
+const CLIENT_CERTIFICATE_MISSING_ERROR_CODE = -200;
+
+class NetworkManager {
+    public clients: Record<string, Client> = {};
```

Comment:
we need a reference for the generic client, don't we? or we import GenericClient somewhere else somehow?